setupext.py

```diff
@@ -353,52 +351,28 @@ def set_pkgconfig_path(self):
         except KeyError:
             os.environ['PKG_CONFIG_PATH'] = pkgconfig_path
 
-    def setup_extension(self, ext, package, default_include_dirs=[],
-                        default_library_dirs=[], default_libraries=[],
-                        alt_exec=None):
-        """
-        Add parameters to the given `ext` for the given `package`.
-        """
-        flag_map = {
-            '-I': 'include_dirs', '-L': 'library_dirs', '-l': 'libraries'}
-
-        executable = alt_exec
-        if self.has_pkgconfig:
-            executable = (self.pkg_config + ' {0}').format(package)
-
-        use_defaults = True
-
-        if executable is not None:
-            command = "{0} --libs --cflags ".format(executable)
-
+    def setup_extension(self, ext, package,
+                        alt_exec=None, default_libraries=()):
+        """Add parameters to the given *ext* for the given *package*."""
+        cmd = ([self.pkg_config, package] if self.pkg_config
+                      else alt_exec)
+        if cmd is not None:
             try:
-                output = subprocess.check_output(
-                    command, shell=True, stderr=subprocess.STDOUT)
-            except subprocess.CalledProcessError:
+                flags = shlex.split(subprocess.check_output(
+                    [*cmd, "--cflags", "--libs"], universal_newlines=True))
```

Comment:
This is potentially a different encoding than the file system encoding used below.

Reply:
good catch, changed to use os.fsdecode() with a comment for clarification.